agent/agent.go

```diff
@@ -602,6 +602,9 @@ func (a *Agent) runTask(ctx context.Context, tcInput *taskContext, nt *apimodels
 	defer span.End()
 	tc.traceID = span.SpanContext().TraceID().String()
 
+	tc.dataDisk, err = dataPartition(ctx)
```

Comment:
Since this is part of task setup, it would be preferable if we could 1. absorb this into `setupTask` for consistency with how the rest of the task config is initialized and 2. write a unit test to protect against regressions. One of the big pain points in the agent refactor was fixing bugs in task setup due to the slow initialization of `tc` over the course of several steps and the fact that the critical setup steps were untested.

Probably same thing with the metrics server setup below this, but I won't argue that strongly for moving a piece of code not directly related to the intended change.

Reply:
moved to setupTask.
I'm unsure about a unit test. To make it actually get the device names for whatever its running on seems fragile. It could be broken up to just test the loop that pulls out device names from the partitions list, if that seems valuable.

Reply:
Hmm, I guess the question here would be, if this functionality broke for some reason (e.g. we upgraded gopsutil, which has in the past broken existing behavior), would anyone notice, ask for a fix, etc.? If it's something that we or users expect to come to rely upon, a simple unit test to verify that it can collect device names would be useful.

Reply:
added a test